File: HOT100/14.贪心算法/55.跳跃游戏.cpp
/*
 * @lc app=leetcode.cn id=55 lang=cpp
 *
 * [55] 跳跃游戏
 */
#include "tools.h"
// @lc code=start
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int co = nums[0];
        for (int i = 0; i <= co; i ++ )
        {
            co = max(co, i + nums[i]);
            if (i >= nums.size() - 1) return 1;
        }
        return 0;
    }
};
// @lc code=end

File: HOT100/14.2/55.跳跃游戏.py
#
# @lc app=leetcode.cn id=55 lang=python3
#
# [55] 跳跃游戏
#
from mytools import *
# @lc code=start
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n == 1: return True
        cur_co = nxt_co = 0
        for i, x in enumerate(nums):
            nxt_co = max(nxt_co, i + x)
            if i == cur_co:
                cur_co = nxt_co
                if cur_co >= n - 1: return True
        return False
# @lc code=end

